src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectGenerator.cs

```diff
@@ -492,7 +475,7 @@ internal bool AreAllTemplatePathsValid(IRunnableProjectConfig templateConfig, Ru
         internal JObject ReadJObjectFromIFile(IFile file)
         {
             using (Stream s = file.OpenRead())
-            using (TextReader tr = new StreamReader(s, true))
+            using (TextReader tr = new StreamReader(s, System.Text.Encoding.UTF8, true))
```

Comment:
Will it be a breaking change for anyone used different encoding now?

Reply:
Apparently, stream reader [already uses UTF8 by default](https://docs.microsoft.com/en-us/dotnet/api/system.io.streamreader.-ctor?view=net-5.0#System_IO_StreamReader__ctor_System_String_), but it failed to read the characters correctly in some of the tests which I suspect are caused by the differences in `StreamReader` in .NET 4.7. If someone was using a different encoding, most likely their template didn't work on other machines anyway.

I think letting users choose their own encoding won't work for us, since these files will need to be read by others who are potentially on machines with different default encodings. We can try to determine the encoding as well. But since the default encoding of json files according to [json specification](https://www.ietf.org/rfc/rfc4627.txt) is already UTF8 and considering how common UTF-8 is, I think it is safe to go with UTF-8 here.